lib/AST/RequirementMachine/RewriteSystem.h

```diff
@@ -165,9 +176,12 @@ class Rule final {
     Permanent = true;
   }
 
-  void markExplicit() {
-    assert(!Explicit && !Permanent &&
+  void markExplicit(Optional<unsigned> requirementID) {
```

Comment:
Since propagateExplicitBits() is wholly separate from propagateRequirementIDs() can we split this up into markExplicit() and setRequirementID()?

The current isExplicit() bit is only used in the minimal conformances algorithm to break ties in some odd cases. I'd like to get rid of it one day.